packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
```

Comment:
Is `fetch` required since we're trying to encapsulate the source of the data?

Plus is it ultimately retrieving contract data?

Perhaps `getNFTContractInfo` so we don't have to interpret what is metadata and what is data?

Reply:
Renamed to `getNFTContractInfo`. 

Concerning `ERC-1155`: Let's assume this is for `ERC-721` for now because I couldn't successfully read the specific collection with an id by using API yet. 

Lets assume we want to get collection details of `0x99a9b7c1116f9ceeb1652de04d5969cce509b069:407000000:407999999`
There is an `id` field in the API specs (https://nft.api.cx.metamask.io/docs#/Account-V1/NftV1Controller_getCollections) but in this case not sure what the id should be.

cc: @sahar-fehri @bergeron

Reply:
The api to getCollections should work for both ERC1155 and ERC721;
The id field should correspond to the contract address for single contract collections ERC1155 or ERC721;
I think you are asking how do you know the id in advance for shared contract addresses?
Usually that collection id is returned after you have already called one of the other APIs like getUserTokens.

I assume in your case you would have in advance the contract address and the tokenId? If that is correct, would calling the [getTokens](https://nft.metafi-dev.codefi.network/docs#/Account-V1/NftV1Controller_getTokens) API instead work for you?
The api will return collection informations, and you can provide your `contractAddress:tokenId` in the `tokens` param. You wont have to worry about what is the collection Id for shared contracts.


For example for this ERC1155 [collection](https://opensea.io/assets/ethereum/0x495f947276749ce646f68ac8c248420045cb7b5e/74857273924061293339526985223456472152818821852613239829843965314197901803521)
you can pass this `0x495f947276749ce646f68ac8c248420045cb7b5e:74857273924061293339526985223456472152818821852613239829843965314197901803521` in the tokens param and it would return the token infos and collection info

Reply:
Changed this to use `getTokens` API.